src/test/kotlin/no/nav/tilleggsstonader/sak/brev/brevmottaker/BrevmottakereServiceTest.kt

```diff
@@ -0,0 +1,212 @@
+package no.nav.tilleggsstonader.sak.brev.brevmottaker
+
+import io.mockk.every
+import io.mockk.mockk
+import no.nav.tilleggsstonader.kontrakter.brevmottaker.BrevmottakerOrganisasjonDto
+import no.nav.tilleggsstonader.kontrakter.brevmottaker.BrevmottakerPersonDto
+import no.nav.tilleggsstonader.kontrakter.brevmottaker.MottakerRolle
+import no.nav.tilleggsstonader.libs.test.assertions.catchThrowableOfType
+import no.nav.tilleggsstonader.sak.behandling.BehandlingService
+import no.nav.tilleggsstonader.sak.behandling.domain.BehandlingStatus
+import no.nav.tilleggsstonader.sak.infrastruktur.database.Sporbar
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.ApiFeil
+import no.nav.tilleggsstonader.sak.util.behandling
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.http.HttpStatus.BAD_REQUEST
+import java.util.*
+
+class BrevmottakereServiceTest {
+
+    private val brevmottakereRepository = mockk<BrevmottakerRepository>()
+    private val behandlingService = mockk<BehandlingService>()
+
+    private val brevmottakereService: BrevmottakereService =
+        BrevmottakereService(brevmottakereRepository, behandlingService)
+
+    val brevmottakerPersonDtoVanligBruker =
+        BrevmottakerPersonDto(UUID.randomUUID(), "12312312323", "Test Testersen", MottakerRolle.BRUKER)
+
+    val brevmottakerOrganisasjonDto =
+        BrevmottakerOrganisasjonDto(UUID.randomUUID(), "45645646456", "Forretninger AS", "Navn navnesen")
+
+    val vanligeBrevmottakereDto =
+        BrevmottakereDto(listOf(brevmottakerPersonDtoVanligBruker), listOf(brevmottakerOrganisasjonDto))
+
+    @Test
+    fun `lagreBrevmottakere skal kaste feilmelding dersom behandligsstatus er FATTER_VEDTAK`() {
+        val behandling = behandling(status = BehandlingStatus.FATTER_VEDTAK)
+
+        every { behandlingService.hentBehandling(any()) } returns behandling
+
+        val feil = catchThrowableOfType<ApiFeil> {
+            brevmottakereService.lagreBrevmottakere(UUID.randomUUID(), vanligeBrevmottakereDto)
+        }
+        assertThat(feil.message).contains("Kan ikke oppdatere brevmottakere fordi behandling er låst for redigering.")
+        assertThat(feil.httpStatus).isEqualTo(BAD_REQUEST)
```

Comment:
Tenker httpStatus her ikke er så viktig å sjekke (gjelder alle tester), så her går det også å bruke `assertThatThrownBy` hvis man ønsker. Den sjekker dog ikke at typen er riktig, men det har ikke så mye å si tenker jeg.

```suggestion
        assertThatThrownBy {
            brevmottakereService.lagreBrevmottakere(UUID.randomUUID(), vanligeBrevmottakereDto)
        }.hasMessageContaining("Kan ikke oppdatere brevmottakere fordi behandling er låst for redigering.")
```